.circleci/config.yml

```diff
@@ -231,23 +231,40 @@ jobs:
 
       - run:
           name: ipynb-check
-          command: ./ci/clean_notebooks.py --check ./docs/tutorials
+          command: pre-commit run --all check-notebooks
 
       - run:
           name: typeignore-check
           command: ./ci/check_typeignore.py ${SRC_FILES}
 
       - run:
           name: flake8
-          command: flake8 --version && flake8 -j "${NUM_CPUS}" ${SRC_FILES}
+          command: |
+            flake8 --version
+            pre-commit run --all flake8
 
       - run:
           name: black
-          command: black --version && black --check --diff ${SRC_FILES}
+          command: |
+            black --version
+            pre-commit run --all black
+            pre-commit run --all black-jupyter
+
+      - run:
+          name: isort
+          command: pre-commit run --all isort
 
       - run:
           name: codespell
-          command: codespell -I .codespell.skip --skip='*.pyc,tests/testdata/*,*.ipynb,*.csv' ${SRC_FILES}
+          command: pre-commit run --all codespell
+
+      - run:
+          name: lint-misc
+          command: |
+            pre-commit run --all check-ast
+            pre-commit run --all trailing-whitespace
+            pre-commit run --all end-of-file-fixer
+            pre-commit run --all check-toml
 
   doctest:
```

Comment:
Should we replace this with `pre-commit run --all docs`?

Reply:
Ignoring per our earlier conversation.